types/address.go

```diff
@@ -55,6 +56,10 @@ func AccAddressFromHex(address string) (addr AccAddress, err error) {
 
 // AccAddressFromBech32 creates an AccAddress from a Bech32 string.
 func AccAddressFromBech32(address string) (addr AccAddress, err error) {
+	if len(strings.TrimSpace(address)) == 0 {
```

Comment:
Are we sure we want to do this direction too (decoding empty strings as valid bech32-addresses)? That seems like it could lead to UX confusion or griefing (tricking users into sending coins to empty addresses, for example).

Reply:
Good point. @sunnya97 what's the new empty address in bech32 ?

Reply:
That should be caught by the `ValidateBasic` of the Msgs.

Reply:
Really, that seems kinda annoying, why not ensure it at the type level instead?

Reply:
If we do it that way *every* `msg.ValidateBasic()` has to remember to check *every* address in the message manually, and if they ever miss one bad things could happen.

Reply:
Doing it at the type level would translate to making all addresses `[tmhash.Size]byte`, and then we have to use pointers in a lot of places in order to avoid re-allocating those 32 bytes (as copying 32 bytes in all the method calls is non-negligible imo). We also lose most of the convenience methods we have on slices. At one point I started working on exactly this PR awhile ago, I think I came to the conclusion that it may in fact be slower, but I wasn't looking at it from a security perspective. If were fine with using pointers all over, then I think its fine to make that type conversion. I agree type system safety checks are better than a "CheckAddress" method everywhere.

Reply:
Lets make a new issue for this though, its basically entirely independent of this PR.

Reply:
I don't quite follow - I'm concerned about this *particular* addition, which will accept `""` (the empty string) as a valid bech32-address. Previously that wasn't the case, and any address when de-serialized in `GetFromBech32` would be automatically checked for the correct length and checksum. Addresses can still be `[]byte`, that's not my concern.

Reply:
i think the more underlying problem is that check tx may pass a tx with invalid address. That should ideally be caught asap. Then any built tx would fail in it's check tx phase.

Reply:
Can we not just do the length check in these function calls (TM address length)?

Reply:
I think we should return an error for invalid addresses here, rather than returning an invalid AccAddress{}.
For simplicity we should use a single global constant tmhash.TruncatedSize (*definitely* not 32 bytes, but 20).
And if the length is invalid, this function should return an error immediately.